apis/duck/v1alpha1/addressable_types.go

```diff
@@ -26,15 +26,17 @@ import (
 
 // Addressable provides a generic mechanism for a custom resource
 // definition to indicate a destination for message delivery.
-// (Currently, only hostname is supported, and HTTP is implied. In the
-// future, additional schemes may be supported, and path components
+// (Currently, only hostname, port and path are supported, and HTTP is implied. In the
```

Comment:
Just curious if instead of adding Port / Path as their separate fields, we should be thinking about adding something like a URI here. So, for hostname, protocol, port and path are fixed, where as all those could be specified in the URI?

Reply:
*url.URL seems like the right candidate here. In fact even with this change I am not sure how to update event sources that we do not own or control. However if we move to url.URL then we will future proof ourselves from now on.
Was wondering why wasnt it dont that way initially (reading the comments in code file)? Just to make sure I am not missing any context before I update it.